modules/integration/tests-ui-integration/src/test/java/org/wso2/carbon/mb/ui/test/dlc/DLCDurableTopicTestCase.java

```diff
@@ -0,0 +1,284 @@
+
+        //Testing restore messages
+        restoringMessageID = dlcContentPage.restoreFunction();
+        waitUntilMessagesReceived(9L);
+
+        //Thread sleep until messages sent to DLC after breaching maximum number of retrying
+        AndesClientUtils.sleepForInterval(80000L);
+
+        // number of messages received by consumer client after restore function triggered.
+        Long afterRestoreMessageReceivedCount = consumerClient.getReceivedMessageCount();
+
+        // This assertion will check if consumer client has received messages messages after
+        // restore function triggered from ui. If it receives messages after restore function triggered
+        // this assertion will be success.
+        Assert.assertTrue(beforeRestoreMessageReceivedCount < afterRestoreMessageReceivedCount,
+                          restoringMessageID + " Durable topic message not successfully restored.");
+    }
+
+    /**
+     * This method will wait until given number of messages received by the consumer client.
+     *
+     * @param numberOfMessages this will specify number of messages to wait before proceed
+     *
+     */
+    public void waitUntilMessagesReceived(Long numberOfMessages) {
+
+
+        while (true) {
+            Long count = consumerClient.getReceivedMessageCount();
+            if (count >= numberOfMessages) {
+                break;
+            }
+            AndesClientUtils.sleepForInterval(1000L);
+        }
+    }
+
+
+    /**
+     * Check whether element is present or not
+     *
+     * @param id which element check for its availability
+     * @return availability of the element
+     */
+    public boolean isElementPresent(String id) {
+        return driver.findElements(By.xpath(id)).size() != 0;
+    }
+
+    /**
+     * Search messageID through all messages in the queue
+     *
+     * @param deletingMessageID - Searching messageID
+     * @param queueName             - Searching queue
+     * @return whether messageID available or not
+     */
+    private boolean checkMessages(String deletingMessageID, String queueName) {
+        boolean isSuccessful = true;
+        if (isElementPresent(UIElementMapper.getInstance()
+                                     .getElement("mb.dlc.browse.content.table"))) {
+            WebElement queueTable = driver.findElement(By.xpath(UIElementMapper.getInstance().
+                    getElement("mb.dlc.browse.content.table")));
+            List<WebElement> rowElementList = queueTable.findElements(By.tagName("tr"));
+            // Go through table rows and find deleted messageID
+            for (WebElement row : rowElementList) {
+                List<WebElement> columnList = row.findElements(By.tagName("td"));
+                // Assumption: there are eleven columns. MessageID is in second column
+                if ((columnList.size() == COLUMN_LIST_SIZE) && columnList.get(MESSAGE_ID_COLUMN)
```

Comment:
Since columnList.size() is returning a primitive type 'int' , flipping would not be needed.